src/core/probe/kernel/inspect.rs

```diff
@@ -16,16 +19,36 @@ pub(super) struct TargetDesc {
 /// Provides helpers to inspect probe related information in the kernel.
 pub(crate) struct Inspector {
     btf: Btf,
+    tracable_funcs: Option<HashSet<String>>,
 }
 
 impl Inspector {
     pub(crate) fn new() -> Result<Inspector> {
-        Ok(Inspector {
+        let mut inspector = Inspector {
             #[cfg(not(test))]
             btf: Btf::from_file("/sys/kernel/btf/vmlinux")?,
             #[cfg(test)]
             btf: Btf::from_file("test_data/vmlinux")?,
-        })
+            tracable_funcs: None,
+        };
+
+        // Not all functions we'll get from BTF/kallsyms are traceable. Use the
+        // following, when available, to narrow down our checks.
+        if let Ok(file) = fs::read_to_string("/sys/kernel/debug/tracing/available_filter_functions")
```

Comment:
Agree. Either that or put `_kprobe` in the function name so it's clear it's a kprobe-only thing because it being part of generic `kernel` could lead to confusion.